sympy/core/core.py

```diff
@@ -60,14 +60,9 @@ def __setattr__(self, name, obj):
     def __delattr__(self, name):
         delattr(self.__class__, name)
 
-#A set containing all SymPy class objects
-all_classes = set()
-
 
 class BasicMeta(type):
-
     def __init__(cls, *args, **kws):
-        all_classes.add(cls)
         cls.__sympy__ = property(lambda self: True)
```

Comment:
This can probably just be replaced with `__sympy__ = True` in the class body of `Basic`.

Reply:
Unfortunately, `ManagedProperties` still has a hefty `__init__`, and I don't know if we can remove that metaclass without changing how the assumptions work. Having it probably affects the import time but I'm unclear just how much it affects runtime performance. I know there are some CPython optimizations that can't be done when you have a metaclass, so it's probably something.

Reply:
We could find ways to move metaclass to `__init_subclass__`
I think that there are advantages about it when we carefully read [PEP487](https://peps.python.org/pep-0487/#easier-inheritance-of-definition-time-behaviour)

Reply:
I had forgotten about that. I think the last time I thought about it it wasn't supported by all Python versions we used. I haven't used it before, but hopefully it can work.